### ISSUE TITLE
live demo

### ISSUE THREAD

**Issue by cleesmith** [cleesmith/escalate#4]
Set up a limited/restricted live demo ... leave it up for a week or so.

**Comment by cleesmith** [comment]
Live demo is at [Escalate](http://104.236.92.144/ "Escalate")

**Comment by miklevin** [comment]
Wow, you are serious about having time and motivation!Sent with SolMail App--------- Original Message ---------
From: chris 
To: cleesmith/escalate 
Date: Fri Jan 02 20:16:42 EST 2015
Subject: Re: [escalate] live demo (#4)

Live demo is at Escalate

**Comment by cleesmith** [comment]
Kids went back to their lives, vacation isn't over until Monday, it's too cold for the neighbor's cats to visit, so it's just me, a fireplace, and a Mac :)   Plus a lot of the code is from other projects.  I'm just hoping it can be useful ... as I feel a lot the common ```seo tasks``` are missing, maybe the experts can contribute ideas at least or code is even better.

**Comment by miklevin** [comment]
Impressive. The seo functions won't be hard and they will be fascinating to do as vids. Ferrets out so many nuanced issues. Especially when we get to the crawls. I'll be interested in looking at your solutions especially in regards to scheduling. Sent with SolMail App--------- Original Message ---------
From: chris 
To: cleesmith/escalate 
Date: Fri Jan 02 20:28:40 EST 2015
Subject: Re: [escalate] live demo (#4)

Kids went back to their lives, vacation isn't over until Monday, it's too cold for the neighbor's cats to visit, so it's just me, a fireplace, and a Mac :)   Plus a lot of the code is from other projects.  I'm just hoping it can be useful ... as I feel a lot the common seo tasks are missing, maybe the experts can contribute ideas at least or code is even better.

**Comment by cleesmith** [comment]
As a ruby/rails programmer ```by day``` I use ```resque``` or ```sidekiq``` which are both persisted using ```redis```, but for escalate I used ```apscheduler``` with no persistence of the jobs scheduled ... they run or are forgotten based on the process running which seemed good enough if one is using the app ```locally```.

**Comment by cleesmith** [comment]
I'm going to try to stay in sync with pipulate, and I probably can if you're only working on Fridays ... otherwise I will fall behind.

**Comment by miklevin** [comment]
I like your thinking. The more we eliminate formal database back end requirements, the more swiss army knife like it's going to be. Sent with SolMail App--------- Original Message ---------
From: chris 
To: cleesmith/escalate 
Date: Fri Jan 02 20:35:35 EST 2015
Subject: Re: [escalate] live demo (#4)

As a ruby/rails programmer by day I use resque or sidekiq which are both persisted using redis, but for escalate I used apscheduler with no persistence of the jobs scheduled ... they run or are forgotten based on the process running which seemed good enough if one is using the app locally.

**Comment by cleesmith** [comment]
I guess it's not ideal to have Java as a dependency for elasticsearch.  Java is the only part that truly requires an installation.  Elasticsearch/Kibana is a ```download and upzip```.  Escalate is ```python+pip+flask+etc``` so that part is the same as pipulate, so it should run on levinux.  I doubt elasticsearch would run on levinux very well, although there is a demo where someone runs an elasticsearch cluster on 5 raspberry pi's ... it was very slow but worked.

**Comment by miklevin** [comment]
Look at the Shove external API. If you use a native shelve database, you can back end it later with larger dbs. Sent with SolMail App--------- Original Message ---------
From: chris 
To: cleesmith/escalate 
Date: Fri Jan 02 20:44:19 EST 2015
Subject: Re: [escalate] live demo (#4)

I guess it's not ideal to have Java as a dependency for elasticsearch.  Java is the only part that truly requires an installation.  Elasticsearch/Kibana is a download and upzip.  Escalate is python+pip+flask+etc so that part is the same as pipulate, so it should run on levinux.  I doubt elasticsearch would run on levinux very well, although there is a demo where someone runs an elasticsearch cluster on 5 raspberry pi's ... it was very slow but worked.

**Comment by cleesmith** [comment]
I will look at Shove, but my issue with anything like sql databases is the lack of full-text search capabilities and aggregations of data.  Some db's claim to search/aggs but when you try them out they fall way short of lucene ... even the mightiest of them, Oracle, falls short.

**Comment by miklevin** [comment]
I hear a lot of good things about the lucene stack. Sent with SolMail App--------- Original Message ---------
From: chris 
To: cleesmith/escalate 
Date: Fri Jan 02 20:53:38 EST 2015
Subject: Re: [escalate] live demo (#4)

I will look at Shove, but my issue with anything like sql databases is the lack of full-text search capabilities and aggregations of data.  Some db's claim to search/aggs but when you try them out they fall way short of lucene ... even the mightiest of them, Oracle, falls short.

**Comment by cleesmith** [comment]
Would you be interested in making a video of escalate ?  I, well, for lack of a better word, suck at youtube videos.

**Comment by miklevin** [comment]
I think so. First, I'll have to "get it". Pipulate with a different interface plus scheduling. Sent with SolMail App--------- Original Message ---------
From: chris 
To: cleesmith/escalate 
Date: Fri Jan 02 21:03:22 EST 2015
Subject: Re: [escalate] live demo (#4)

Would you be interested in making a video of escalate ?  I, well, for lack of a better word, suck at youtube videos.

**Comment by cleesmith** [comment]
Cool ... or you can wait until there are more seo functions that you/others would find useful.
I'm not looking for any benefit from this project myself, it was just something new/different to work on that might be useful to others ... I guess the direction it goes(or doesn't) is up to the seo type users.

**Comment by cleesmith** [comment]
Forgot to ask, do you have any recommendations for learning more about seo, well, outside of pipulate.
I did try to mimic spreadsheets by using DataTables to present the data and search results, but it would be nice to know a bit more about seo other than the experts use excel or google spreadsheets ... there are a few of those tutorials.

**Comment by cleesmith** [comment]
Live demo was moved to port 5000:
http://104.236.92.144:5000/
so a simple kibana dashboard could be shown.

**Comment by miklevin** [comment]
Your amount of work on this, and the user interface work on it is amazing. I'd like to hear more.

For SEO, just grab the contents of the title tag, meta description, and headline elements. And of course the URL. The function I'm going to write soon will look for the keyword(s) that appear to be targeted. The articles on moz.com are a good place to start learning about SEO.

**Comment by cleesmith** [comment]
I just found moz this morning, but thanks.
I did add some basic kibana dashboards to the demo.  Crawlers do grab the title tag, meta description/keywords tags, the first h1 tag, and the url for the page ... just not sure what to do with it, so
I simply show it.  It also strips the page of html tags, stop words, stores that as plain text, and display the word frequencies within the plain text.

Next up, I guess, is tackling user funcs and a generic way to present them ... too bad folks don't like to look at json because that's what elasticsearch returns without any custom coding, oh well.

I look forward to the new seo function(s).
